src/mui/list/List.js

```diff
@@ -37,6 +37,13 @@ class List extends Component {
         }
     }
 
+    componentWillUpdate(nextProps) {
+        const changedProps = this.getChangedProps(nextProps);
+        Object.keys(changedProps).forEach(changedPropName => {
+            this.setFilter(changedPropName, changedProps[changedPropName]);
```

Comment:
You're doing a `setFilter` for *all* props... some props don't relate to filters